lbrynet/lbrynet_daemon/Downloader.py

```diff
@@ -49,17 +49,16 @@ def __init__(self, sd_identifier, session, wallet, lbry_file_manager, exchange_r
         self.timeout_counter = 0
         self.code = None
         self.sd_hash = None
-        self.wallet = wallet
         self.session = session
+        self.wallet = self.session.wallet
         self.exchange_rate_manager = exchange_rate_manager
         self.payment_rate_manager = self.session.payment_rate_manager
-        self.lbry_file_manager = lbry_file_manager
         self.sd_identifier = sd_identifier
         self.downloader = None
         self.checker = LoopingCall(self.check_status)
 
         # fired when the download is complete
-        self.finished_deferred = defer.Deferred(None)
+        self.finished_deferred = None
```

Comment:
Yes. Note that finished_deferred is set to downloader.start() at the end of download() function. 

Meaning that this initialized defer doesn't do anything. And if you try to attach a callback to it before downloader.start() is called, which someone did in the Daemon, it never gets called back.